rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/helmfile/HelmfileTemplateUtils.java

```diff
@@ -0,0 +1,157 @@
+    if ("git/repo".equals(artifactType)) {
+      env.downloadArtifactTarballAndExtract(artifactDownloader, helmfileTemplateArtifact);
+
+      // If there's no helmfile path specified, assume it lives in the root of
+      // the git/repo artifact.
+      helmfileFilePath =
+          env.resolvePath(Optional.ofNullable(request.getHelmfileFilePath()).orElse(""));
+    } else {
+      try {
+        helmfileFilePath = downloadArtifactToTmpFile(env, helmfileTemplateArtifact);
+      } catch (SpinnakerHttpException e) {
+        throw new SpinnakerHttpException(fetchFailureMessage("template", e), e);
+      } catch (IOException | SpinnakerException e) {
+        throw new IllegalStateException(fetchFailureMessage("template", e), e);
+      }
+    }
+
+    log.info("path to helmfile: {}", helmfileFilePath);
+
+    try {
+      // not a stream to keep exception handling cleaner
+      for (Artifact valueArtifact : inputArtifacts.subList(1, inputArtifacts.size())) {
+        valuePaths.add(downloadArtifactToTmpFile(env, valueArtifact));
+      }
+    } catch (SpinnakerHttpException e) {
+      throw new SpinnakerHttpException(fetchFailureMessage("values file", e), e);
+    } catch (IOException | SpinnakerException e) {
+      throw new IllegalStateException(fetchFailureMessage("values file", e), e);
+    }
+
+    List<String> command = new ArrayList<>();
+    String executable = helmfileConfigurationProperties.getExecutablePath();
+
+    command.add(executable);
+    command.add("template");
+    command.add("--file");
+    command.add(helmfileFilePath.toString());
+
+    command.add("--helm-binary");
+    command.add(getHelm3ExecutablePath());
+
+    String environment = request.getEnvironment();
+    if (environment != null && !environment.isEmpty()) {
+      command.add("--environment");
+      command.add(environment);
+    }
+
+    String namespace = request.getNamespace();
+    if (namespace != null && !namespace.isEmpty()) {
+      command.add("--namespace");
+      command.add(namespace);
+    }
+
+    if (request.isIncludeCRDs()) {
+      command.add("--include-crds");
+    }
+
+    Map<String, Object> overrides = request.getOverrides();
+    if (!overrides.isEmpty()) {
+      List<String> overrideList = new ArrayList<>();
+      for (Map.Entry<String, Object> entry : overrides.entrySet()) {
+        overrideList.add(entry.getKey() + "=" + entry.getValue().toString());
+      }
+      command.add("--set");
+      command.add(String.join(",", overrideList));
+    }
+
+    if (!valuePaths.isEmpty()) {
+      command.add("--values");
+      command.add(valuePaths.stream().map(Path::toString).collect(Collectors.joining(",")));
+    }
+
+    result.setCommand(command);
+
+    return result;
+  }
+
+  private String fetchFailureMessage(String description, Exception e) {
+    return "Failed to fetch helmfile " + description + ": " + e.getMessage();
+  }
+
+  public String removeTestsDirectoryTemplates(String inputString) {
```

Comment:
Let's figure out a way to reduce the duplication / share the code in [HelmTemplateUtils](https://github.com/spinnaker/rosco/blob/d49a43507428ebd67bd0942f1050c755e45ab6bc/rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/helm/HelmTemplateUtils.java#L141-L153).

Reply:
Moved this method under an abstract class.